darshan-util/pydarshan/tests/test_data_access_by_filesystem.py

```diff
@@ -0,0 +1,462 @@
+                            filesystem_roots
+                         """, [
+
+     (pd.Series([1], index=pd.Index(['/p'], name='filesystem_root'), name='filepath'),
+      pd.Series([1], index=pd.Index(['/p'], name='filesystem_root'), name='filepath'),
+      pd.Series([1.049e+6], index=pd.Index(['/p'], name='filesystem_root'), name='POSIX_BYTES_READ'),
+      pd.Series([1.049e+6], index=pd.Index(['/p'], name='filesystem_root'), name='POSIX_BYTES_WRITTEN'),
+      ['/p'],
+         ),
+     # test case where files read/written are zero
+     (pd.Series([0], index=pd.Index(['/p'], name='filesystem_root'), name='filepath'),
+      pd.Series([0], index=pd.Index(['/p'], name='filesystem_root'), name='filepath'),
+      # NOTE: very strange to be able to read/write bytes to 
+      # a filesystem and yet have no files read or written
+      # to on that filesystem (this might be an error someday?)
+      # see comment:
+      # https://github.com/darshan-hpc/darshan/pull/397#discussion_r683621305
+      pd.Series([1.049e+6], index=pd.Index(['/p'], name='filesystem_root'), name='POSIX_BYTES_READ'),
+      pd.Series([1.049e+6], index=pd.Index(['/p'], name='filesystem_root'), name='POSIX_BYTES_WRITTEN'),
+      ['/p'],
+         ),
+       ])
+def test_plot_data_labels(file_rd_series, file_wr_series, bytes_rd_series, bytes_wr_series, filesystem_roots):
+    # regression test for label spacing in plot
+    # based on review comment in gh-397
+    fig = plt.figure()
+    data_access_by_filesystem.plot_data(fig=fig,
+                                        file_rd_series=file_rd_series,
+                                        file_wr_series=file_wr_series,
+                                        bytes_rd_series=bytes_rd_series,
+                                        bytes_wr_series=bytes_wr_series,
+                                        filesystem_roots=filesystem_roots)
+    for ax in fig.axes:
+        for child in ax.get_children():
+            if isinstance(child, matplotlib.text.Text):
+                actual_text = child.get_text()
+                if actual_text not in ['/p', '']:
+                    # count the leading spaces for each label
+                    leading_spaces = len(actual_text) - len(actual_text.lstrip(' '))
+                    # check there is always 1 leading space for each label
+                    assert leading_spaces == 1
+
+
+
+def test_plot_data_shared_x_axis():
+    # regression test for case described here:
+    # https://github.com/darshan-hpc/darshan/pull/397#pullrequestreview-717403104
+    # https://github.com/darshan-hpc/darshan/pull/397#issuecomment-889504530
+    filesystem_roots = ['/usr', '/yellow', '/green', '/global']
+    rd_bytes = [1e7, 1e8, 1e9, 1e10]
+    wr_bytes = [1e8, 1e9, 1e10, 1e11]
+    rd_file_cts = [1e3, 1e4, 1e5, 1e6]
+    wr_file_cts = [1e2, 1e3, 1e4, 1e5]
+    # multiply by the MiB conversion factor to get round numbers in the output
+    factor = 1.049e+6
```

Comment:
This should be updated to the same value we are using for the MiB conversion factor (`factor = 1048576`).